internal/scheduler/nodedb/nodedb.go

```diff
@@ -878,3 +901,31 @@ func (nodeDb *NodeDb) stringFromPodRequirementsNotMetReason(reason schedulerobje
 		return s
 	}
 }
+
+func (nodeDb *NodeDb) nodeDbKeyFromNode(out []byte, node *schedulerobjects.Node, priority int32) []byte {
```

Comment:
Previously, we relied on memdb methods for constructing a combined key from the node nodeType and resources. However, this caused lots of unnecessary allocs. Now we create the key ourselves using this method instead.